Convert markdown issue templates to forms

gravityfargo opened the pull request:
This PR converts the traditional markdown issue template to the beta forms template. The documentation can be found [here](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms). 

You can find the live version of what the form looks like for users on my fork and prefilled issues I submitted to show you the result of those forms. 

I also added the discord link as an item in the [template picker page](https://github.com/gravityfargo/liquidctl/issues/new/choose). 


[bug report form](https://github.com/gravityfargo/liquidctl/issues/new?assignees=&labels=bug&template=bug_report.yml)
[example issue](https://github.com/gravityfargo/liquidctl/issues/1)

[new device form](https://github.com/gravityfargo/liquidctl/issues/new?assignees=&labels=new+device&template=new-device.yml)
[example issue](https://github.com/gravityfargo/liquidctl/issues/2)

[other issues form](https://github.com/gravityfargo/liquidctl/issues/new?assignees=&labels=question&template=help-request.yml)
[example issue](https://github.com/gravityfargo/liquidctl/issues/3)

jonasmalacofilho commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,61 @@
+name: Bug Report
+description: Report an issue with existing supported devices.
+labels: ["bug"]
+body:
```
The order in which each information is requested was intentional: a good description of the bug is the most important part, followed by the output from all relevant commands, and then the additional information.

gravityfargo replied:
I moved the entries around to reflect your comment and the current template.

jonasmalacofilho commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,61 @@
+    id: info-device
+    attributes:
+      label: Affected Device
+      description: What is the name of the device you are using that is not functioning as expected? If more than one devices are experiencing issues list them here separated by a comma.
```
This is unnecessary long, IMO:

```suggestion
      description: Which device isn't functioning as expected?
```

jonasmalacofilho commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,61 @@
+      label: Affected Device
+      description: What is the name of the device you are using that is not functioning as expected? If more than one devices are experiencing issues list them here separated by a comma.
+      placeholder: e.g. Corsair Hydro H100i Pro XT
+    validations:
+      required: true
```
Some bugs aren't specific to any particular device, so I don't think this should be required.

jonasmalacofilho commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,61 @@
+      placeholder: e.g. Corsair Hydro H100i Pro XT
+    validations:
+      required: true
+  - type: input
+    id: info-distro
+    attributes:
+      label: Linux Distribution
+      description: Which Linux distribution are you using?
+      placeholder: ex. Arch
+    validations:
+      required: true
```
liquidctl supports platforms other than Linux (even though that happens to be my main/preferred one).

jonasmalacofilho commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,61 @@
+      placeholder: ex. Arch
+    validations:
+      required: true
+  - type: input
+    id: install-method
+    attributes:
+      label: Installation Method
+      description: How did you install Liquidctl?
+      placeholder: e.g. Official Arch Linux repository 
+    validations:
+      required: true
+  - type: input
+    id: info-version
+    attributes:
+      label: Liquidctl Version
+      description: Output of `liquidctl --version`
+      placeholder: e.g. liquidctl v1.9.0 (Linux-5.16.16-arch1-1-x86_64-with-glibc2.35) 
+    validations:
+      required: true
```
Not a major thing, but can you make both descriptions account for the scenario where the bug is in the installation procedure?

(But keeping them short & simple for the far more common case of a bug after the installation was successful).

gravityfargo replied:
I moved this code around. I made a new comment highlighting the changes.

jonasmalacofilho commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,61 @@
+  - type: textarea
+    id: user-solutions
+    attributes:
+      label: Command output with debug flag
```
```suggestion
      label: Output of all relevant commands (with `--debug` flag)
```

jonasmalacofilho commented on .github/ISSUE_TEMPLATE/config.yml:
```diff
@@ -1 +1,5 @@
 blank_issues_enabled: false
+contact_links:
+  - name: Discord
+    url: https://discord.gg/GyCBjQhqCd
+    about: "Ask for support in our Discord Server!."
```
I rather not forward users to Discord, which turns into a mess for anything that might might eventually be turned into an issue.

We have discussions enabled in this repository, that's probably a better place to redirect these types of questions.

jonasmalacofilho commented on .github/ISSUE_TEMPLATE/help-request.yml:
```diff
@@ -0,0 +1,11 @@
+name: Other questions
+description: For all other questions / issues
+labels: ["question"]
+body:
+  - type: textarea
+    id: question-body
+    attributes:
+      label: Describe your question.
+      description: Please provide any relevant information
+    validations:
+      required: true
```
The wording should relate to issues and, more generally, requests (as in feature requests), but _not_ questions. The issue tracker is not a place for questions.

gravityfargo replied:
I deleted this form entirely and added a link to the discussions where I had put the discord link

jonasmalacofilho reviewed (commented):
This is something I've been meaning to do for a while. Thanks for looking into it!

Besides the specific comments, please also mind the capitalization of "liquidctl" (lowercase, unless at the start of a sentence, but try to avoid that) and of labels in general (I don't have a strong preference, just be consistent regarding whether only the first or all words are title cased).

gravityfargo commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -58,4 +43,37 @@ body:
+      placeholder: e.g. Arch, macOS, Windows 10
+    validations:
+      required: true
+  - type: input
+    id: install-method
+    attributes:
+      label: Installation method
+      description: How did you install liquidctl? If the install failed please indicate what methods you tried.
+      placeholder: e.g. Official Arch Linux repository 
+    validations:
+      required: true
+  - type: input
+    id: info-version
+    attributes:
+      label: Version of liquidctl
+      description: Output of `liquidctl --version` If the install failed, indicate the version you tried to install.
+      placeholder: e.g. liquidctl v1.9.0 (Linux-5.16.16-arch1-1-x86_64-with-glibc2.35) 
+    validations:
+      required: true
```
I couldn't think of a more concise way of putting this.

jonasmalacofilho replied:
It's fine!

gravityfargo commented:
Updated live examples in my Repo:

Replaced Discord with Discussion [template picker page](https://github.com/gravityfargo/liquidctl/issues/new/choose). 

[bug report form](https://github.com/gravityfargo/liquidctl/issues/new?assignees=&labels=bug&template=bug_report.yml)
[example issue](https://github.com/gravityfargo/liquidctl/issues/4)

[new device form](https://github.com/gravityfargo/liquidctl/issues/new?assignees=&labels=new+device&template=new-device.yml)
[example issue](https://github.com/gravityfargo/liquidctl/issues/5)

jonasmalacofilho commented:
I might take another look later, and eventually tweak a thing or two (e.g. which fields are required). But it's certainly good enough to merge, and an improvement over our current templates.

Thanks!
